token/services/owner/manager.go

```diff
@@ -50,99 +43,66 @@ func NewManager(tmsProvide TokenManagementServiceProvider, np NetworkProvider, t
 }
 
 // Owner returns the Owner for the given TMS
-func (cm *Manager) Owner(tms *token.ManagementService) (*Owner, error) {
+func (cm *Manager) Owner(tmsID token.TMSID) (*Owner, error) {
 	cm.mutex.Lock()
 	defer cm.mutex.Unlock()
 
-	id := tms.ID().String()
+	id := tmsID.String()
 	logger.Debugf("get ttxdb for [%s]", id)
 	c, ok := cm.owners[id]
 	if !ok {
 		// add an entry
-		if err := cm.storage.Put(tms.ID(), ""); err != nil {
-			return nil, errors.Wrapf(err, "failed to store db entry in KVS [%s]", tms.ID())
+		if err := cm.storage.Put(tmsID, ""); err != nil {
+			return nil, errors.Wrapf(err, "failed to store db entry in KVS [%s]", tmsID)
 		}
 		var err error
-		c, err = cm.newOwner(tms)
+		c, err = cm.newOwner(tmsID)
 		if err != nil {
-			return nil, errors.WithMessagef(err, "failed to instantiate owner for wallet [%s]", tms.ID())
+			return nil, errors.WithMessagef(err, "failed to instantiate owner for wallet [%s]", tmsID)
 		}
 		cm.owners[id] = c
 	}
 	return c, nil
 }
 
-func (cm *Manager) Restore() error {
-	logger.Infof("restore owner dbs...")
-	it, err := cm.storage.Iterator()
-	if err != nil {
-		return errors.WithMessagef(err, "failed to list existing owners")
-	}
-	defer func(it storage.Iterator[*storage.DBEntry]) {
-		err := it.Close()
-		if err != nil {
-			logger.Warnf("failed to close iterator [%s]", err)
-		}
-	}(it)
-	for {
-		if !it.HasNext() {
-			logger.Infof("restore owner dbs...done")
-			return nil
-		}
-		entry, err := it.Next()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get next entry")
-		}
-		logger.Infof("restore owner dbs for entry [%s]...", entry.TMSID.String())
-		tms, err := cm.tmsProvider.GetManagementService(token.WithTMSID(entry.TMSID))
-		if err != nil {
-			return errors.WithMessagef(err, "cannot find TMS [%s]", entry.TMSID)
-		}
-		if err := cm.restore(tms); err != nil {
-			return errors.Errorf("cannot bootstrap auditdb for [%s]", entry.TMSID)
-		}
-		logger.Infof("restore owner dbs for entry [%s]...done", entry.TMSID.String())
-	}
-}
-
-func (cm *Manager) newOwner(tms *token.ManagementService) (*Owner, error) {
-	db, err := cm.ttxdbProvider.DB(&tmsWallet{tms: tms})
+func (cm *Manager) newOwner(tmsID token.TMSID) (*Owner, error) {
+	db, err := cm.ttxdbProvider.DBByTMSId(tmsID)
 	if err != nil {
-		return nil, errors.WithMessagef(err, "failed to get ttxdb for [%s:%s]", tms.ID().Network, tms.ID().Channel)
+		return nil, errors.WithMessagef(err, "failed to get ttxdb for [%s:%s]", tmsID.Network, tmsID.Channel)
```

Comment:
you don't have the namespace. you could use directly "[%s]", tmsID, since it implements `String()`